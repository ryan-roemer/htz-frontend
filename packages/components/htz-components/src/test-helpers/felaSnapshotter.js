import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
