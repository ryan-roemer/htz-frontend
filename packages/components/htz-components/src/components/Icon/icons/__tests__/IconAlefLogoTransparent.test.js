
/* global window */
/* *************************************************************** *
 * THIS IS AN AUTO GENERATED FILE. PLEASE DO NOT EDIT IT DIRECTLY.
 *
 * If you want to change the styleguide example, it is generated
 * from the `embedsFileTemplate.js` file is this directory.
 * *************************************************************** */
import React from 'react';
import felaSnapshotter from '../../../../test-helpers/felaSnapshotter';
import IconAlefLogoTransparent from '../IconAlefLogoTransparent';

describe('<IconAlefLogoTransparent />', () => {
  /* The error tests are left as documentation, but are commented out for the redability of the test results */
  // describe('Errors', () => {
  //   it('throw an error when the value passed to the "size" prop isn't a number', () => {
  //     expect(() => felaSnapshotter(<IconAlefLogoTransparent size="2rem" />)).toThrow(
  //       /^An Icon's "size" prop may only be passed a "number"/
  //     );
  //   });
  //   it('throw an error when the value passed to the "color" prop isn't a named color (string)', () => {
  //     expect(() => felaSnapshotter(<IconAlefLogoTransparent color="red" />)).toThrow(
  //       /is not a named-color/
  //     );
  //   });
  //   it('throw an error when the value passed to the "color" prop isn't a named color (array of strings)', () => {
  //     expect(() =>
  //       felaSnapshotter(<IconAlefLogoTransparent color={[ 'primary', 'nope', ]} />)
  //     ).toThrow(/is not a named-color/);
  //   });
  //   it('throw an error when the value passed to the "fill" prop isn't a named color (string)', () => {
  //     expect(() => felaSnapshotter(<IconAlefLogoTransparent fill="red" />)).toThrow(
  //       /is not a named-color/
  //     );
  //   });
  //   it('throw an error when the value passed to the "fill" prop isn't a named color (array of strings)', () => {
  //     expect(() =>
  //       felaSnapshotter(<IconAlefLogoTransparent fill={[ 'primary', 'nope', ]} />)
  //     ).toThrow(/is not a named-color/);
  //   });
  // });
  describe('DOM element', () => {
    testCase('render correctly when no props are passed', <IconAlefLogoTransparent />);
    testCase(
      'pass attributes defined in the "attrs" prop to the DOM element',
      <IconAlefLogoTransparent attrs={{ 'aria-hidden': true, }} />
    );
    testCase('add "id" attribute to DOM element', <IconAlefLogoTransparent id="hasId" />);
    testCase(
      'add "onClick" attribute to DOM element',
      <IconAlefLogoTransparent id={e => console.log('a')} />
    );
  });

  describe('styles', () => {
    testCase("set icon's color", <IconAlefLogoTransparent color={[ 'primary', 'base', ]} />);
    testCase(
      "set icon's color responsively",
      <IconAlefLogoTransparent
        color={[
          { until: 'xl', value: [ 'primary', 'base', ], },
          { from: 'xl', value: [ 'bodyText', 'base', ], },
        ]}
      />
    );
    testCase("set icon's fill", <IconAlefLogoTransparent fill={[ 'primary', 'base', ]} />);
    testCase(
      "set icon's fill responsively",
      <IconAlefLogoTransparent
        fill={[
          { until: 'xl', value: [ 'primary', 'base', ], },
          { from: 'xl', value: [ 'bodyText', 'base', ], },
        ]}
      />
    );
    testCase("set icon's size", <IconAlefLogoTransparent size={6} />);
    testCase(
      "set icon's size responsively",
      <IconAlefLogoTransparent
        size={[ { until: 'xl', value: 6, }, { from: 'xl', value: 7, }, ]}
      />
    );
    testCase(
      'set misc styles on an icon',
      <IconAlefLogoTransparent miscStyles={{ position: 'relative', }} />
    );
    testCase(
      'responsively set misc styles on an icon',
      <IconAlefLogoTransparent
        miscStyles={{
          position: [
            { until: 'xl', value: 'relative', },
            { from: 'xl', value: 'static', },
          ],
        }}
      />
    );
  });
});

function testCase(name, Component, { only = false, } = {}) {
  const testFunc = only ? it.only : it;
  testFunc(name, () => {
    const { component, styles, } = felaSnapshotter(Component);
    expect(component).toMatchSnapshot();
    expect(styles).toMatchSnapshot();
  });
}
