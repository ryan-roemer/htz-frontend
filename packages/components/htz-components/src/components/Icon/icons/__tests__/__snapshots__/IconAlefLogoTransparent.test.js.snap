// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconAlefLogoTransparent /> DOM element add "id" attribute to DOM element 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    id="hasId"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 8.5a119.5 119.5 0 1 0 0 239 119.5 119.5 0 0 0 0-239zm69.3 182.1c-.3 0-.5.2-.5.5L90.2 112.6a21.1 21.1 0 0 0-7.9 15.4c-.9 6.8-.7 14.1.7 21.7 1.4 7.7 5.7 11.5 12.9 11.5h19.2c-1 12.4.7 23.4 5.1 32.7h-51c-.7-.6-.8-1.4-.5-2.3a171.7 171.7 0 0 0 6.5-36.9c.3-6-.2-11.6-1.4-17.1a75.1 75.1 0 0 1-1.9-17.5c0-5.6.3-9.7.9-12.4.7-2.7 1.3-7.3 1.9-13.8A53.4 53.4 0 0 0 69.5 64c0-.3.2-.5.5-.5h8.4c6.2 5 15.5 12.1 28 21.3l33.7 25a9 9 0 0 0 3.5-3.7c1.1-1.9 2.3-4.7 3.5-8.4 1.2-4.7 1.2-8.6 0-11.7a23.8 23.8 0 0 1-1.9-9.3c-.3-3.1-.3-5.7.2-7.7s.7-3.8.7-5.4h43.5c-5.6 9-5.4 21.3.5 36.9h-30.9c-2.2 0-4.2.9-6.1 2.8a14.4 14.4 0 0 0-3.7 7c-.3 1.8-.1 3.8.7 5.8.8 2.1 1.7 3.4 2.6 4 8.1 5.7 15 10.6 20.6 15a268 268 0 0 0 17.8 12.6c-.9 4.7-1.4 8.2-1.6 10.5a56.2 56.2 0 0 0 3.6 21.1 86.5 86.5 0 0 0 4.2 11.3z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlefLogoTransparent /> DOM element add "id" attribute to DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconAlefLogoTransparent /> DOM element add "onClick" attribute to DOM element 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    id={[Function]}
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 8.5a119.5 119.5 0 1 0 0 239 119.5 119.5 0 0 0 0-239zm69.3 182.1c-.3 0-.5.2-.5.5L90.2 112.6a21.1 21.1 0 0 0-7.9 15.4c-.9 6.8-.7 14.1.7 21.7 1.4 7.7 5.7 11.5 12.9 11.5h19.2c-1 12.4.7 23.4 5.1 32.7h-51c-.7-.6-.8-1.4-.5-2.3a171.7 171.7 0 0 0 6.5-36.9c.3-6-.2-11.6-1.4-17.1a75.1 75.1 0 0 1-1.9-17.5c0-5.6.3-9.7.9-12.4.7-2.7 1.3-7.3 1.9-13.8A53.4 53.4 0 0 0 69.5 64c0-.3.2-.5.5-.5h8.4c6.2 5 15.5 12.1 28 21.3l33.7 25a9 9 0 0 0 3.5-3.7c1.1-1.9 2.3-4.7 3.5-8.4 1.2-4.7 1.2-8.6 0-11.7a23.8 23.8 0 0 1-1.9-9.3c-.3-3.1-.3-5.7.2-7.7s.7-3.8.7-5.4h43.5c-5.6 9-5.4 21.3.5 36.9h-30.9c-2.2 0-4.2.9-6.1 2.8a14.4 14.4 0 0 0-3.7 7c-.3 1.8-.1 3.8.7 5.8.8 2.1 1.7 3.4 2.6 4 8.1 5.7 15 10.6 20.6 15a268 268 0 0 0 17.8 12.6c-.9 4.7-1.4 8.2-1.6 10.5a56.2 56.2 0 0 0 3.6 21.1 86.5 86.5 0 0 0 4.2 11.3z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlefLogoTransparent /> DOM element add "onClick" attribute to DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconAlefLogoTransparent /> DOM element pass attributes defined in the "attrs" prop to the DOM element 1`] = `
<div>
  <svg
    aria-hidden={true}
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 8.5a119.5 119.5 0 1 0 0 239 119.5 119.5 0 0 0 0-239zm69.3 182.1c-.3 0-.5.2-.5.5L90.2 112.6a21.1 21.1 0 0 0-7.9 15.4c-.9 6.8-.7 14.1.7 21.7 1.4 7.7 5.7 11.5 12.9 11.5h19.2c-1 12.4.7 23.4 5.1 32.7h-51c-.7-.6-.8-1.4-.5-2.3a171.7 171.7 0 0 0 6.5-36.9c.3-6-.2-11.6-1.4-17.1a75.1 75.1 0 0 1-1.9-17.5c0-5.6.3-9.7.9-12.4.7-2.7 1.3-7.3 1.9-13.8A53.4 53.4 0 0 0 69.5 64c0-.3.2-.5.5-.5h8.4c6.2 5 15.5 12.1 28 21.3l33.7 25a9 9 0 0 0 3.5-3.7c1.1-1.9 2.3-4.7 3.5-8.4 1.2-4.7 1.2-8.6 0-11.7a23.8 23.8 0 0 1-1.9-9.3c-.3-3.1-.3-5.7.2-7.7s.7-3.8.7-5.4h43.5c-5.6 9-5.4 21.3.5 36.9h-30.9c-2.2 0-4.2.9-6.1 2.8a14.4 14.4 0 0 0-3.7 7c-.3 1.8-.1 3.8.7 5.8.8 2.1 1.7 3.4 2.6 4 8.1 5.7 15 10.6 20.6 15a268 268 0 0 0 17.8 12.6c-.9 4.7-1.4 8.2-1.6 10.5a56.2 56.2 0 0 0 3.6 21.1 86.5 86.5 0 0 0 4.2 11.3z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlefLogoTransparent /> DOM element pass attributes defined in the "attrs" prop to the DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconAlefLogoTransparent /> DOM element render correctly when no props are passed 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 8.5a119.5 119.5 0 1 0 0 239 119.5 119.5 0 0 0 0-239zm69.3 182.1c-.3 0-.5.2-.5.5L90.2 112.6a21.1 21.1 0 0 0-7.9 15.4c-.9 6.8-.7 14.1.7 21.7 1.4 7.7 5.7 11.5 12.9 11.5h19.2c-1 12.4.7 23.4 5.1 32.7h-51c-.7-.6-.8-1.4-.5-2.3a171.7 171.7 0 0 0 6.5-36.9c.3-6-.2-11.6-1.4-17.1a75.1 75.1 0 0 1-1.9-17.5c0-5.6.3-9.7.9-12.4.7-2.7 1.3-7.3 1.9-13.8A53.4 53.4 0 0 0 69.5 64c0-.3.2-.5.5-.5h8.4c6.2 5 15.5 12.1 28 21.3l33.7 25a9 9 0 0 0 3.5-3.7c1.1-1.9 2.3-4.7 3.5-8.4 1.2-4.7 1.2-8.6 0-11.7a23.8 23.8 0 0 1-1.9-9.3c-.3-3.1-.3-5.7.2-7.7s.7-3.8.7-5.4h43.5c-5.6 9-5.4 21.3.5 36.9h-30.9c-2.2 0-4.2.9-6.1 2.8a14.4 14.4 0 0 0-3.7 7c-.3 1.8-.1 3.8.7 5.8.8 2.1 1.7 3.4 2.6 4 8.1 5.7 15 10.6 20.6 15a268 268 0 0 0 17.8 12.6c-.9 4.7-1.4 8.2-1.6 10.5a56.2 56.2 0 0 0 3.6 21.1 86.5 86.5 0 0 0 4.2 11.3z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlefLogoTransparent /> DOM element render correctly when no props are passed 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconAlefLogoTransparent /> styles responsively set misc styles on an icon 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d htz-e"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 8.5a119.5 119.5 0 1 0 0 239 119.5 119.5 0 0 0 0-239zm69.3 182.1c-.3 0-.5.2-.5.5L90.2 112.6a21.1 21.1 0 0 0-7.9 15.4c-.9 6.8-.7 14.1.7 21.7 1.4 7.7 5.7 11.5 12.9 11.5h19.2c-1 12.4.7 23.4 5.1 32.7h-51c-.7-.6-.8-1.4-.5-2.3a171.7 171.7 0 0 0 6.5-36.9c.3-6-.2-11.6-1.4-17.1a75.1 75.1 0 0 1-1.9-17.5c0-5.6.3-9.7.9-12.4.7-2.7 1.3-7.3 1.9-13.8A53.4 53.4 0 0 0 69.5 64c0-.3.2-.5.5-.5h8.4c6.2 5 15.5 12.1 28 21.3l33.7 25a9 9 0 0 0 3.5-3.7c1.1-1.9 2.3-4.7 3.5-8.4 1.2-4.7 1.2-8.6 0-11.7a23.8 23.8 0 0 1-1.9-9.3c-.3-3.1-.3-5.7.2-7.7s.7-3.8.7-5.4h43.5c-5.6 9-5.4 21.3.5 36.9h-30.9c-2.2 0-4.2.9-6.1 2.8a14.4 14.4 0 0 0-3.7 7c-.3 1.8-.1 3.8.7 5.8.8 2.1 1.7 3.4 2.6 4 8.1 5.7 15 10.6 20.6 15a268 268 0 0 0 17.8 12.6c-.9 4.7-1.4 8.2-1.6 10.5a56.2 56.2 0 0 0 3.6 21.1 86.5 86.5 0 0 0 4.2 11.3z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlefLogoTransparent /> styles responsively set misc styles on an icon 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}@media (max-width: 79.9375em){.htz-d{position:relative}}@media (min-width: 80em){.htz-e{position:static}}"`;

exports[`<IconAlefLogoTransparent /> styles set icon's color 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 8.5a119.5 119.5 0 1 0 0 239 119.5 119.5 0 0 0 0-239zm69.3 182.1c-.3 0-.5.2-.5.5L90.2 112.6a21.1 21.1 0 0 0-7.9 15.4c-.9 6.8-.7 14.1.7 21.7 1.4 7.7 5.7 11.5 12.9 11.5h19.2c-1 12.4.7 23.4 5.1 32.7h-51c-.7-.6-.8-1.4-.5-2.3a171.7 171.7 0 0 0 6.5-36.9c.3-6-.2-11.6-1.4-17.1a75.1 75.1 0 0 1-1.9-17.5c0-5.6.3-9.7.9-12.4.7-2.7 1.3-7.3 1.9-13.8A53.4 53.4 0 0 0 69.5 64c0-.3.2-.5.5-.5h8.4c6.2 5 15.5 12.1 28 21.3l33.7 25a9 9 0 0 0 3.5-3.7c1.1-1.9 2.3-4.7 3.5-8.4 1.2-4.7 1.2-8.6 0-11.7a23.8 23.8 0 0 1-1.9-9.3c-.3-3.1-.3-5.7.2-7.7s.7-3.8.7-5.4h43.5c-5.6 9-5.4 21.3.5 36.9h-30.9c-2.2 0-4.2.9-6.1 2.8a14.4 14.4 0 0 0-3.7 7c-.3 1.8-.1 3.8.7 5.8.8 2.1 1.7 3.4 2.6 4 8.1 5.7 15 10.6 20.6 15a268 268 0 0 0 17.8 12.6c-.9 4.7-1.4 8.2-1.6 10.5a56.2 56.2 0 0 0 3.6 21.1 86.5 86.5 0 0 0 4.2 11.3z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlefLogoTransparent /> styles set icon's color 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:#0B7EB5}.htz-c{fill:#FFF}"`;

exports[`<IconAlefLogoTransparent /> styles set icon's color responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 8.5a119.5 119.5 0 1 0 0 239 119.5 119.5 0 0 0 0-239zm69.3 182.1c-.3 0-.5.2-.5.5L90.2 112.6a21.1 21.1 0 0 0-7.9 15.4c-.9 6.8-.7 14.1.7 21.7 1.4 7.7 5.7 11.5 12.9 11.5h19.2c-1 12.4.7 23.4 5.1 32.7h-51c-.7-.6-.8-1.4-.5-2.3a171.7 171.7 0 0 0 6.5-36.9c.3-6-.2-11.6-1.4-17.1a75.1 75.1 0 0 1-1.9-17.5c0-5.6.3-9.7.9-12.4.7-2.7 1.3-7.3 1.9-13.8A53.4 53.4 0 0 0 69.5 64c0-.3.2-.5.5-.5h8.4c6.2 5 15.5 12.1 28 21.3l33.7 25a9 9 0 0 0 3.5-3.7c1.1-1.9 2.3-4.7 3.5-8.4 1.2-4.7 1.2-8.6 0-11.7a23.8 23.8 0 0 1-1.9-9.3c-.3-3.1-.3-5.7.2-7.7s.7-3.8.7-5.4h43.5c-5.6 9-5.4 21.3.5 36.9h-30.9c-2.2 0-4.2.9-6.1 2.8a14.4 14.4 0 0 0-3.7 7c-.3 1.8-.1 3.8.7 5.8.8 2.1 1.7 3.4 2.6 4 8.1 5.7 15 10.6 20.6 15a268 268 0 0 0 17.8 12.6c-.9 4.7-1.4 8.2-1.6 10.5a56.2 56.2 0 0 0 3.6 21.1 86.5 86.5 0 0 0 4.2 11.3z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlefLogoTransparent /> styles set icon's color responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-d{fill:#FFF}@media (max-width: 79.9375em){.htz-b{color:#0B7EB5}}@media (min-width: 80em){.htz-c{color:#2D2D2D}}"`;

exports[`<IconAlefLogoTransparent /> styles set icon's fill 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 8.5a119.5 119.5 0 1 0 0 239 119.5 119.5 0 0 0 0-239zm69.3 182.1c-.3 0-.5.2-.5.5L90.2 112.6a21.1 21.1 0 0 0-7.9 15.4c-.9 6.8-.7 14.1.7 21.7 1.4 7.7 5.7 11.5 12.9 11.5h19.2c-1 12.4.7 23.4 5.1 32.7h-51c-.7-.6-.8-1.4-.5-2.3a171.7 171.7 0 0 0 6.5-36.9c.3-6-.2-11.6-1.4-17.1a75.1 75.1 0 0 1-1.9-17.5c0-5.6.3-9.7.9-12.4.7-2.7 1.3-7.3 1.9-13.8A53.4 53.4 0 0 0 69.5 64c0-.3.2-.5.5-.5h8.4c6.2 5 15.5 12.1 28 21.3l33.7 25a9 9 0 0 0 3.5-3.7c1.1-1.9 2.3-4.7 3.5-8.4 1.2-4.7 1.2-8.6 0-11.7a23.8 23.8 0 0 1-1.9-9.3c-.3-3.1-.3-5.7.2-7.7s.7-3.8.7-5.4h43.5c-5.6 9-5.4 21.3.5 36.9h-30.9c-2.2 0-4.2.9-6.1 2.8a14.4 14.4 0 0 0-3.7 7c-.3 1.8-.1 3.8.7 5.8.8 2.1 1.7 3.4 2.6 4 8.1 5.7 15 10.6 20.6 15a268 268 0 0 0 17.8 12.6c-.9 4.7-1.4 8.2-1.6 10.5a56.2 56.2 0 0 0 3.6 21.1 86.5 86.5 0 0 0 4.2 11.3z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlefLogoTransparent /> styles set icon's fill 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#0B7EB5}"`;

exports[`<IconAlefLogoTransparent /> styles set icon's fill responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 8.5a119.5 119.5 0 1 0 0 239 119.5 119.5 0 0 0 0-239zm69.3 182.1c-.3 0-.5.2-.5.5L90.2 112.6a21.1 21.1 0 0 0-7.9 15.4c-.9 6.8-.7 14.1.7 21.7 1.4 7.7 5.7 11.5 12.9 11.5h19.2c-1 12.4.7 23.4 5.1 32.7h-51c-.7-.6-.8-1.4-.5-2.3a171.7 171.7 0 0 0 6.5-36.9c.3-6-.2-11.6-1.4-17.1a75.1 75.1 0 0 1-1.9-17.5c0-5.6.3-9.7.9-12.4.7-2.7 1.3-7.3 1.9-13.8A53.4 53.4 0 0 0 69.5 64c0-.3.2-.5.5-.5h8.4c6.2 5 15.5 12.1 28 21.3l33.7 25a9 9 0 0 0 3.5-3.7c1.1-1.9 2.3-4.7 3.5-8.4 1.2-4.7 1.2-8.6 0-11.7a23.8 23.8 0 0 1-1.9-9.3c-.3-3.1-.3-5.7.2-7.7s.7-3.8.7-5.4h43.5c-5.6 9-5.4 21.3.5 36.9h-30.9c-2.2 0-4.2.9-6.1 2.8a14.4 14.4 0 0 0-3.7 7c-.3 1.8-.1 3.8.7 5.8.8 2.1 1.7 3.4 2.6 4 8.1 5.7 15 10.6 20.6 15a268 268 0 0 0 17.8 12.6c-.9 4.7-1.4 8.2-1.6 10.5a56.2 56.2 0 0 0 3.6 21.1 86.5 86.5 0 0 0 4.2 11.3z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlefLogoTransparent /> styles set icon's fill responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}@media (max-width: 79.9375em){.htz-c{fill:#0B7EB5}}@media (min-width: 80em){.htz-d{fill:#2D2D2D}}"`;

exports[`<IconAlefLogoTransparent /> styles set icon's size 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 8.5a119.5 119.5 0 1 0 0 239 119.5 119.5 0 0 0 0-239zm69.3 182.1c-.3 0-.5.2-.5.5L90.2 112.6a21.1 21.1 0 0 0-7.9 15.4c-.9 6.8-.7 14.1.7 21.7 1.4 7.7 5.7 11.5 12.9 11.5h19.2c-1 12.4.7 23.4 5.1 32.7h-51c-.7-.6-.8-1.4-.5-2.3a171.7 171.7 0 0 0 6.5-36.9c.3-6-.2-11.6-1.4-17.1a75.1 75.1 0 0 1-1.9-17.5c0-5.6.3-9.7.9-12.4.7-2.7 1.3-7.3 1.9-13.8A53.4 53.4 0 0 0 69.5 64c0-.3.2-.5.5-.5h8.4c6.2 5 15.5 12.1 28 21.3l33.7 25a9 9 0 0 0 3.5-3.7c1.1-1.9 2.3-4.7 3.5-8.4 1.2-4.7 1.2-8.6 0-11.7a23.8 23.8 0 0 1-1.9-9.3c-.3-3.1-.3-5.7.2-7.7s.7-3.8.7-5.4h43.5c-5.6 9-5.4 21.3.5 36.9h-30.9c-2.2 0-4.2.9-6.1 2.8a14.4 14.4 0 0 0-3.7 7c-.3 1.8-.1 3.8.7 5.8.8 2.1 1.7 3.4 2.6 4 8.1 5.7 15 10.6 20.6 15a268 268 0 0 0 17.8 12.6c-.9 4.7-1.4 8.2-1.6 10.5a56.2 56.2 0 0 0 3.6 21.1 86.5 86.5 0 0 0 4.2 11.3z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlefLogoTransparent /> styles set icon's size 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}.htz-d{font-size:6rem}"`;

exports[`<IconAlefLogoTransparent /> styles set icon's size responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d htz-e"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 8.5a119.5 119.5 0 1 0 0 239 119.5 119.5 0 0 0 0-239zm69.3 182.1c-.3 0-.5.2-.5.5L90.2 112.6a21.1 21.1 0 0 0-7.9 15.4c-.9 6.8-.7 14.1.7 21.7 1.4 7.7 5.7 11.5 12.9 11.5h19.2c-1 12.4.7 23.4 5.1 32.7h-51c-.7-.6-.8-1.4-.5-2.3a171.7 171.7 0 0 0 6.5-36.9c.3-6-.2-11.6-1.4-17.1a75.1 75.1 0 0 1-1.9-17.5c0-5.6.3-9.7.9-12.4.7-2.7 1.3-7.3 1.9-13.8A53.4 53.4 0 0 0 69.5 64c0-.3.2-.5.5-.5h8.4c6.2 5 15.5 12.1 28 21.3l33.7 25a9 9 0 0 0 3.5-3.7c1.1-1.9 2.3-4.7 3.5-8.4 1.2-4.7 1.2-8.6 0-11.7a23.8 23.8 0 0 1-1.9-9.3c-.3-3.1-.3-5.7.2-7.7s.7-3.8.7-5.4h43.5c-5.6 9-5.4 21.3.5 36.9h-30.9c-2.2 0-4.2.9-6.1 2.8a14.4 14.4 0 0 0-3.7 7c-.3 1.8-.1 3.8.7 5.8.8 2.1 1.7 3.4 2.6 4 8.1 5.7 15 10.6 20.6 15a268 268 0 0 0 17.8 12.6c-.9 4.7-1.4 8.2-1.6 10.5a56.2 56.2 0 0 0 3.6 21.1 86.5 86.5 0 0 0 4.2 11.3z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlefLogoTransparent /> styles set icon's size responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}@media (max-width: 79.9375em){.htz-d{font-size:6rem}}@media (min-width: 80em){.htz-e{font-size:7rem}}"`;

exports[`<IconAlefLogoTransparent /> styles set misc styles on an icon 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M128 8.5a119.5 119.5 0 1 0 0 239 119.5 119.5 0 0 0 0-239zm69.3 182.1c-.3 0-.5.2-.5.5L90.2 112.6a21.1 21.1 0 0 0-7.9 15.4c-.9 6.8-.7 14.1.7 21.7 1.4 7.7 5.7 11.5 12.9 11.5h19.2c-1 12.4.7 23.4 5.1 32.7h-51c-.7-.6-.8-1.4-.5-2.3a171.7 171.7 0 0 0 6.5-36.9c.3-6-.2-11.6-1.4-17.1a75.1 75.1 0 0 1-1.9-17.5c0-5.6.3-9.7.9-12.4.7-2.7 1.3-7.3 1.9-13.8A53.4 53.4 0 0 0 69.5 64c0-.3.2-.5.5-.5h8.4c6.2 5 15.5 12.1 28 21.3l33.7 25a9 9 0 0 0 3.5-3.7c1.1-1.9 2.3-4.7 3.5-8.4 1.2-4.7 1.2-8.6 0-11.7a23.8 23.8 0 0 1-1.9-9.3c-.3-3.1-.3-5.7.2-7.7s.7-3.8.7-5.4h43.5c-5.6 9-5.4 21.3.5 36.9h-30.9c-2.2 0-4.2.9-6.1 2.8a14.4 14.4 0 0 0-3.7 7c-.3 1.8-.1 3.8.7 5.8.8 2.1 1.7 3.4 2.6 4 8.1 5.7 15 10.6 20.6 15a268 268 0 0 0 17.8 12.6c-.9 4.7-1.4 8.2-1.6 10.5a56.2 56.2 0 0 0 3.6 21.1 86.5 86.5 0 0 0 4.2 11.3z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconAlefLogoTransparent /> styles set misc styles on an icon 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}.htz-d{position:relative}"`;
